### DISCUSSION TITLE
Quarto possibly overwriting svg font settings in typst output?

### DISCUSSION THREAD

**Discussion by sblgpln** [quarto-dev/quarto-cli#10354] [Q&A] [answered]
### Description

I am creating a batch of reports in quarto, using the jupyter engine to process python code and typst to typset and format the text. 
I want to create images using matlplotlib, with specific font settings (size and weight) and have them display in the document. The output .svg files behave as they should, but the displayed image in the pdf document has font size and defaulting to what looks like regular weight font. I tried to override this by setting `svg.fonttype` to `none` instead of `path` in matplotlib but this is not ideal.

Would appreciate any pointers on what could be going on. I am on Quarto v 1.5.54 in jupyter lab, running Ubuntu-22.04 on wsl.

Here is some example code. I have included some text to check that the necessary static font files are in the system.

`````qmd

---
format: typst

execute:
  keep-ipynb: true
  keep-typ: true
  echo: false
---

````{=typst}
#set text(11pt, font: "IBM Plex Sans", weight:"light") // 

== Some text

#let weights = (
  "thin",
  "extralight",
  "light",
  "regular",
  "medium",
  "semibold",
  "bold",
  "extrabold",
  "black",
)

#for w in weights {
  [
    #w
    #h(1fr)
    #text(weight: w, lorem(10))

  ]
}

````

````{python}
import matplotlib.pyplot as plt
import numpy as np
plt.rcParams['font.weight']='light'
plt.rcParams['axes.titleweight'] ='semibold'
plt.rcParams['svg.fonttype']='none'


# Data for plotting
t = np.arange(0.0, 2.0, 0.01)
s = 1 + np.sin(2 * np.pi * t)

fig, ax = plt.subplots()
ax.plot(t, s)

ax.set(xlabel='X', ylabel='Y')
ax.grid()

plt.title('A Plot', fontsize= 24, fontweight = 'semibold')
plt.savefig('plot.svg')
plt.show()
````

![Saved plot - quarto](plot.svg)

````{=typst}
#figure(
  image("plot.svg"),
  caption: [Saved plot - typst]
)
````

`````

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.


Did you check your SVG itself?

**Reply by sblgpln** [reply]
Yes, the svg is displaying as intended.
https://github.com/quarto-dev/quarto-cli/discussions/10354#discussioncomment-10123828

**Comment by cscheid** [comment]
This isn't Quarto - take your python code cell, name it as `test.py`, and run it. Then, open your .svg file in a text editor. This is what I see:

```
...
      <text style="font: 200 10px 'DejaVu Sans', 'Bitstream Vera Sans', 'Computer Modern Sans Serif', 'Lucida Grande', 'Verdana', 'Geneva', 'Lucid', 'Arial', 'Helvetica', 'Avant Garde', sans-serif; text-anchor: middle" x="73.832727" y="322.182437" transform="rotate(-0 73.832727 322.182437)">0.00</text>
...
```

**Reply by sblgpln** [reply]
Hi @cscheid, thank you for your response!

I didn't add the specific font to the code to keep the code small, but the size and weight issue still persists. For the axis markings the svg says `text style="font: 200 10px ...` and for the title near the end it says `text style="font: 600 24px ...` which is what I passed to matplotlib, but in the displayed image in the compiled document they look the same. If I open the svg file in an image viewer it looks as intended.  

````
...
    <text style="font: 600 24px 'DejaVu Sans', 'Bitstream Vera Sans', 'Computer Modern Sans Serif', 'Lucida Grande', 'Verdana', 'Geneva', 'Lucid', 'Arial', 'Helvetica', 'Avant Garde', sans-serif; text-anchor: middle" x="230.625" y="24.24" transform="rotate(-0 230.625 24.24)">A Plot</text>
...

````

**Reply by cscheid** [reply]
@sblgpln I don't understand what you're referring to. Let me emphasize that my point is simply that this cannot possibly be Quarto related (and so we can't help you), because the font specifications appear when running your code from pure Python.

**Comment by gordonwoodhull** [comment, marked as answer]
I did a quick search and I think you may be running into this issue

https://github.com/typst/typst/issues/3162

As a workaround, I
* rendered with `keep-typ:true`
* modified the svg by hand to replace the `font` shorthand with `font-family`, `font-weight`, and `font-size` (as suggested in https://github.com/typst/typst/issues/3162#issuecomment-1908607617)
* ran `quarto typst compile` manually
 
I think this is closer to the desired look:

<img width="423" alt="image" src="https://github.com/user-attachments/assets/ad1a16c0-ee2c-43f3-ab29-8f3cce27db6f">

It looks like they have fixed this on Typst main but they haven't made a new release yet. If they do, we should pick it up our next release.

Anyway as @cscheid said, it's an upstream Typst issue and not really Quarto.

**Reply by sblgpln** [reply]
Thanks, this looks like it!
